frontend/javascripts/oxalis/view/layouting/tracing_layout_view.tsx

```diff
@@ -194,13 +194,16 @@ class TracingLayoutView extends React.PureComponent<PropsWithRouter, State> {
     app.vent.emit("rerender");
 
     if (model != null) {
-      this.setState({
-        model,
-      });
-    }
-
-    if (this.props.autoSaveLayouts) {
-      this.saveCurrentLayout(layoutName);
+      this.setState(
+        {
+          model,
+        },
+        () => {
+          if (this.props.autoSaveLayouts) {
+            this.saveCurrentLayout(layoutName);
```

Comment:
`this.saveCurrentLayout` accessed `this.state.model` and persists it, although the state update did not go through yet. I have no idea how this ever worked before :upside_down_face: